Projects/rym/rick and morty/src/App.jsx

```diff
@@ -0,0 +1,52 @@
+import SearchInput from "./components/search-input/search-input"
+import CharactersList from "./components/characters-list/characters-list"
+import getCharacters from "./services/get_characters";
+import { useState,useEffect } from "react";
+
+function useCharacters() {
+  const [characters, setCharacters] = useState([])
+
+  const searchCharacters = (value='') => getCharacters().then(data => {
+        const charactersFilter = data.results.filter(character =>
+          character.name.toLowerCase().includes(value.toLowerCase())
+        )
+        setCharacters(charactersFilter)
+      }).catch(error => {
+          console.error('Error al obtener los personajes:', error)
+    })
+
+    return {
+      searchCharacters,
+      characters
+    }
+}
+
+function App() {
+  const [searchValue, setSearchValue] = useState('')
+  const {searchCharacters,characters} = useCharacters()
+  useEffect(() => {
+    searchCharacters(searchValue)
+  },[searchValue])
```

Comment:
It means that every time that the input change, are you going to request alllllllll characters? it is too expensive

How can you improve it?

Reply:
can we see this point together?

my idea is i have all characters when is mounted all the page and save them in a array and let it be consult alone this array

Reply:
You have the idea, what should we see together?